api/src/opentrons/protocol_engine/execution/labware_movement.py

```
@@ -89,8 +89,25 @@ async def move_labware_with_gripper(
     ) -> None:
         """Move a loaded labware from one location to another using gripper."""
         use_virtual_gripper = self._state_store.config.use_virtual_gripper
+
         if use_virtual_gripper:
+            # During Analysis we will pass in hard coded estimates for certain positions only accessible during execution
+            # Estimated gripper homed Z position: 166.125 mm
+            attached_tips = self._state_store.pipettes.get_all_attached_tips()
+            if attached_tips:
+                for pipette_id, tip in attached_tips:
+                    if not self._state_store.geometry.validate_gripper_labware_tip_collision(
+                        gripper_homed_position_z=166.125,
+                        tip=tip,
+                        pipette_id=pipette_id,
+                        labware_id=labware_id,
+                        current_location=current_location,
+                    ):
+                        raise LabwareMovementNotAllowedError(
+                            f"Cannot move labware '{self._state_store.labware.get(labware_id).loadName}' when {int(tip.volume)} ul tips are attached."
```

Continuing my comment on the validation function, if that function raises error instead of returning true/ false, you won't need to raise this error twice in this method. So essentially, this new code block reduces to just a call to `validate_gripper_labware_tip_collision`.